### PR TITLE
Updated openai to 1.11.0 and Assistant Demo code

### PR DESCRIPTION
**Pull request by chaseez**

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "command_channels": [
-    1192895746672951340
+    1194820338933043271
```
Comment (byubean) [inline config.json:3]: Don't change the config

File: poetry.lock
Comment (byubean) [inline poetry.lock:1]: Don't check in poetry.lock

File: rubber_duck.py [hunk 2]
```diff
@@ -133,12 +139,12 @@ async def have_conversation(self, thread_id: int, engine: str, prompt: str, init
...
         # Replaces _get_response
         async with self._typing(thread_id):
-            completion: OpenAIObject = await openai.ChatCompletion.acreate(
+            completion: ChatCompletion = await client.chat.completions.create(
                 model=engine,
                 messages=message_history
             )
             logging.debug(f"Completion: {completion}")
-            return completion.choices, completion.usage
+            return completion.dict()
```
Comment (byubean) [inline rubber_duck.py:150]: return the choices and usage dictionaries like we did before. Those sub objects should also have a .dict() method. Then the return type and calling code doesn't need to be different.

File: scratch/Demo-file-objects/bitbot.mdx.json
Comment (byubean) [inline scratch/Demo-file-objects/bitbot.mdx.json:1]: Leave your scratch files out of this commit

**Review by byubean — CHANGES_REQUESTED** [review]
See comments.

**Comment by byubean** [comment]
Also, be sure to merge master into your branch first and resolve the conflicts:
```
git pull origin master
```

File: rubber_duck.py (hunk 2 above)
Comment (chaseez) [reply rubber_duck.py:150]: I can do that for usage, but not for choices. `completion.choices` returns a list, not a dictionary. I'll fiddle around with it and see what I can do
[completion.json](https://github.com/beanlab/rubber-duck/files/14254728/completion.json)

Comment (chaseez) [reply rubber_duck.py]: Ok, I was able to parse out the list of choices and usage as a dictionary before returning and that worked when I ran it on my end

File: my_config.json
Comment (byubean) [inline my_config.json:1]: Don't check in your personal config.

File: rubber_duck.py [hunk 3]
```diff
@@ -113,7 +119,6 @@ async def have_conversation(self, thread_id: int, engine: str, prompt: str, init
                     response = response_message['content'].strip()
 
                     await self._metrics_handler.record_usage(guild_id, thread_id, user_id,
-                                                             engine,
```
Comment (byubean) [inline rubber_duck.py:116]: I want the engine recorded in the usage

**Review by byubean — APPROVED** [review]

**Pull request merged**